controllers/cluster_controller_phases.go

```diff
@@ -88,13 +95,17 @@ func (r *ClusterReconciler) reconcileExternal(ctx context.Context, cluster *clus
 		UID:        cluster.UID,
 	}
 
-	if !util.HasOwnerRef(obj.GetOwnerReferences(), ownerRef) {
-		obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), ownerRef))
-		if err := r.Client.Patch(ctx, obj, objPatch); err != nil {
-			return nil, errors.Wrapf(err,
-				"failed to set OwnerReference on %v %q for Cluster %q in namespace %q",
-				obj.GroupVersionKind(), ref.Name, cluster.Name, cluster.Namespace)
-		}
+	// Add ownerRef to object.
+	obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), ownerRef))
+	if err := r.Client.Patch(ctx, obj, objPatch); err != nil {
```

Comment:
Given that we're patching with the helper, this should be removed

Reply:
Ahh thank you for the pointer. I wanted to make sure not to lose any data before patching.
ptal